### ISSUE TITLE
Missing language chooser on login modal and registration form

### ISSUE THREAD

**Issue by felixhahnweilheim** [humhub/humhub#5426]
### Is your feature request related to a problem? Please describe.

Yes, the language chooser (shown on the separate login page) is missing on the login modal window (shown when guest access is enabled) and on the registration form.

### Describe the solution you'd like

Add the language chooser (`<?= humhub\widgets\LanguageChooser::widget(); ?>`) to
- `/protected/humhub/modules/user/views/auth/login_modal.php`  and
- `/protected/humhub/modules/user/views/registration/index.php`

### Describe alternatives you've considered

**Comment by luke-** [comment]
We could show the Language Chooser somewhere in the Footer Navigation when guest access is enabled

**Comment by funkycram** [comment]
Yes. And I think it should be shown also at the top of the registration form.

**Comment by atdqm** [comment]
yes, in a multilingual context the possibility to choose the language on the registration form is of great interest to us

**Comment by luke-** [comment]
Maybe we could add this to the footer?

**Comment by ArchBlood** [comment]
> Maybe we could add this to the footer?

I would like to also reference the following if this is done;

https://github.com/humhub-contrib/legal/pull/34#issuecomment-1278806197

**Comment by codiflow** [comment]
Is there anything which prevents this simple solution from being integrated into the next version? I think this would be of great help for all international HumHub installations :partying_face:

**Comment by yurabakhtin** [comment]
@luke- We have implemented the language switcher by the module "Translator" [here](https://github.com/humhub/translator/pull/57), but it has been done only for registered users:

![change-lang](https://github.com/user-attachments/assets/79cdd7e7-baad-474c-ba62-07d9c65c6caa)

I have reviewed the code and it can be easy done for guests as well, if we will add the small code on save the form:

```php
        if (Yii::$app->user->isGuest) {
            $languageChooser = new ChooseLanguage();
            $languageChooser->language = $this->language;
            return $languageChooser->save();
        }
```

So it will work like this:

https://github.com/user-attachments/assets/9a5cbf4b-f9f8-4527-8f1f-965a69ad7ee5

Do you agree this solution or we need to implement this for core side in the footer?

If core side, what branch `master` or `develop`?
Also I see some pages like Dashboard, Spaces, People are scrolled down to next pages, so user may not available to the footer until the last page reached, so maybe better place is in the sidebar where the text "Powered by HumHub" is located.

**Comment by luke-** [comment]
Hmm, the solution from the Translator Module is rather specific and for guests we currently have no notification area where we could add the icon.

I would therefore prefer Marc's alt solution and also display the language switch in the footer, as on the login page.
![image](https://github.com/user-attachments/assets/551717a3-7738-4960-a4c4-e5277d14fc31)

We need to find a solution for the problem that the footer is not displayed on some pages (People, CP) anyway.

**Comment by yurabakhtin** [comment]
@luke- Is it normal to keep [here](https://github.com/humhub/humhub/commit/ad29c9d388a6660d9bf5e1250ff8089cba5435b4#diff-998090a04bca64de4ffafc51198af4263186be08fbaa5b52be871d15327edfe1R19) English text?

![de-en](https://github.com/user-attachments/assets/03514ba8-de21-4563-878f-55c08186564f)

**Comment by marc-farre** [comment]
I would say it's good in case the current language is unknown by the user, to understand he can change the language here, as English is the  most known language in the world.

**Comment by yurabakhtin** [comment]
> I would therefore prefer Marc's alt solution and also display the language switch in the footer, as on the login page.

@luke- PR https://github.com/humhub/humhub/pull/7384:

![lang-chooser](https://github.com/user-attachments/assets/402eec39-1890-4370-a7e3-e5acaaeb19b0)

> I would say it's good in case the current language is unknown by the user, to understand he can change the language here, as English is the most known language in the world.

@marc-farre Ok, no problem if it was done specially, I was just wondering why only German language has it.

**Comment by marc-farre** [comment]
I see. It shouldn't be for German only... It means it has been added to the German translation. @luke- what do you think?

**Comment by luke-** [comment]
@marc-farre Regarding the German Translation I've created an Issue  here: https://github.com/humhub/wording/issues/175

Is the PR ok for you?

**Comment by marc-farre** [comment]
Yes, it's fine, thanks.

But I would also add it to the main footer. E.g. with the Calendar module:

![image](https://github.com/user-attachments/assets/1b695f14-9ee7-4fb3-b405-cd901d92fa63)

I've created a commit for it: https://github.com/humhub/humhub/pull/7384/commits/cd588c3d881b9c2fc22fbd38620664de45821692

**Comment by luke-** [comment]
@marc-farre :+1: 

@yurabakhtin Any reason why you skipped the main footer?

**Comment by yurabakhtin** [comment]
> Any reason why you skipped the main footer?

@luke- Sorry, I didn't understand that I should do that, because the pages Spaces and People haven't the footer menu as the Global Calendar page has it.
If we will add the same code `FooterMenu::widget(['location' => FooterMenu::LOCATION_FULL_PAGE])` for the Spaces and People pages then:

![last-page-footer](https://github.com/user-attachments/assets/e6c0b7af-39d2-4650-b491-8911ca9bd323)

Should I do that? Or should we find another solution?

@marc-farre Thank you for [the changes](https://github.com/humhub/humhub/commit/cd588c3d881b9c2fc22fbd38620664de45821692), but it seems the file `static/js/humhub-app.js.tmp` has been added by mistake.

**Comment by marc-farre** [comment]
@yurabakhtin Sorry, I removed the file in a new commit.

For the footer page locations, perhaps we could create a new issue.
Because it's not only a problem with the language chooser, it's also that we cannot access the legal pages from the Space page when not logged in (People is disabled for guests, and when logged in, it's added to the account menu).
Same issue with custom pages.

Once the footer is displayed on all pages, the language chooser will also be available everywhere.
So I would say this PR is fine now.

**Comment by luke-** [comment]
I've created an issue for People/Spaces here: https://github.com/humhub/humhub/issues/7385
Thanks!
